var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -92,6 +95,52 @@ class Perl(Package):  # Perl doesn't use Autotools, it should subclass Package
 
     phases = ['configure', 'build', 'install']
 
+    @classmethod
+    def determine_spec_details(cls, prefix, exes_in_prefix):
+        exe_to_path = dict(
+            (os.path.basename(p), p) for p in exes_in_prefix
+        )
+        if 'perl' not in exe_to_path:
+            return None
+
+        perl = spack.util.executable.Executable(exe_to_path['perl'])
+        output = perl('--version', output=str)
+        version_str = ''
+        shared = False
+        threads = False
+        cpanm = False
+        if output:
```

Comment:
seems like this could just be

```python
if not output:
    return None
```

then dedent the `True` block below.

Reply:
I didn't de-indent because I wanted to keep the output/match parallelism below, but I can be swayed.  Does the new commit look clearer?

Reply:
looks good to me! It's just a nitpick anyway. I tend to write guards and try to avoid ifs with two branches if I can.